src/mono/mono/component/CMakeLists.txt

```diff
@@ -78,17 +81,51 @@ set(${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}-dependencies
   ${MONO_DIAGNOSTICS_TRACING_COMPONENT_NAME}-gen-sources
 )
 
+# marshal_ilgen
+list(APPEND components
+  ${MONO_MARSHAL_ILGEN_COMPONENT_NAME}
+)
+list(APPEND components_for_aot
+  ${MONO_MARSHAL_ILGEN_COMPONENT_NAME}
+)
+
+set(${MONO_MARSHAL_ILGEN_COMPONENT_NAME}-sources
+  ${MONO_COMPONENT_PATH}/marshal_ilgen.c
```

Comment:
Mono traditionally uses - as a separator in file names not _.

Reply:
I was under the impression that the component name "marshal_ilgen" also gets expanded into C variable, so I can't put a dash in it. Maybe I am mistaken and I can change the name of the file independently; I will try that.

Reply:
It looks like it can be made to build with a - instead of a _, but all the other component names (and files with component names in them) use _. This also makes it consistent with the _ for component names  the C code. Should I use _ to be consistent with the other components? Or change it to be -?

Reply:
let's use `-`. we can cleanup the other components file names (but probably not the component names - they're used by the SDKs) in a separate PR. (I think it's probably my fault - I named `hot_reload` like that because it matched the symbol name)

Reply:
Ok, done.